pandas/io/parquet.py

```diff
@@ -53,9 +53,15 @@ def validate_dataframe(df: DataFrame):
         if not isinstance(df, DataFrame):
             raise ValueError("to_parquet only supports IO with DataFrames")
 
-        # must have value column names (strings only)
-        if df.columns.inferred_type not in {"string", "empty"}:
-            raise ValueError("parquet must have string column names")
+        # must have value column names for all index levels (strings only)
+        if df.columns.nlevels > 1:
```

Comment:
If you replace this check with isinstance(df.columns, MultiIndex) that might silence mypy

Reply:
Thanks @dsaxton for the suggestion - looks like replacing the check with `isinstance(df.columns, MultiIndex)` did the trick with silencing mypy on my dev environment (and should do the trick on CI too)!

Reply:
Great, glad it helped